gnosis_vpn-lib/src/balance.rs

```diff
@@ -64,6 +68,20 @@ pub struct PreSafe {
     pub node_wxhopr: Balance<WxHOPR>,
 }
 
+impl PreSafe {
+    pub async fn fetch(priv_key: &ChainKeypair, rpc_provider: &str, node_address: Address) -> Result<Self, Error> {
```

Comment:
This type of interface will make it difficult to integrate the edge client with blokli.

Reply:
Can be rewritten to integrate blokli as needed. We can worry about this once we see an actual interface that we need to integrate